### PR TITLE
more strict 10

### PR DESCRIPTION
**Pull request by chrisnojima**
This mostly changes the typing of our style helpers. before it just cast everything so you'd lose a ton of info. Instead we actually plumb through the objects through things like platformStyles. This lets us pass style objects to text/icon/etc that have restrictions and catch it when there's a real issues instead of having to cast all the time

### CODE REVIEW

File: shared/app/runtime-stats/index.tsx [hunk 1]
```diff
@@ -279,6 +279,7 @@ const styles = Styles.styleSheetCreate(() => ({
   container: Styles.platformStyles({
     common: {
       backgroundColor: Styles.globalColors.blackOrBlack,
+      foo: 'bar',
```
Comment (chrisnojima) [inline shared/app/runtime-stats/index.tsx]: - [ ] check extra

File: shared/styles/colors.tsx [hunk 2]
```diff
@@ -166,7 +166,7 @@ export const colors = {
     return this.white
   },
   get whiteOrGreenDark() {
-    return this.greenDark
+    return this.white
```
Comment (chrisnojima) [inline shared/styles/colors.tsx]: i believe this was a bug ,cc: @adamjspooner

File: shared/styles/css.d.ts [hunk 3]
```diff
@@ -4,8 +4,81 @@ import {StyleProp, ViewStyle, TextStyle, ImageStyle} from 'react-native'
 export type Color = null | string
 type _StylesDesktopOverride = {
   wordBreak?: 'normal' | 'break-all' | 'keep-all' | 'inherit' | 'initial' | 'unset' | 'break-word'
+  WebkitAppRegion?: 'drag' | 'no-drag'
 }
-export type _StylesDesktop = CSSProperties & _StylesDesktopOverride
```
Comment (chrisnojima) [inline shared/styles/css.d.ts]: this style has every conceivable web style, 99% of which we don't use, so in order to speed things up and aid readability we just pull the small subset we use

**Comment by chrisnojima** [comment]
@keybase/react-hackers bump

File: shared/wallets/airdrop/banner/index.tsx [hunk 4]
```diff
@@ -79,7 +79,7 @@ const markdownOverride = {
 
 const styles = Styles.styleSheetCreate(() => ({
   button: Styles.platformStyles({
-    isElectron: Styles.desktopStyles.windowDraggingClickable,
+    isElectron: {...Styles.desktopStyles.windowDraggingClickable},
```
Comment (jakob223) [inline shared/wallets/airdrop/banner/index.tsx]: why do you need to make a new object?

File: shared/styles/shared.tsx [hunk 5]
```diff
@@ -79,20 +79,32 @@ const unifyStyles = (s: any) => ({
...
-  isElectron?: _StylesDesktop
-  // TODO any for now, but we need the types to be handled differently
-}): any => ({
-  ...(options.common ? unifyStyles(options.common) : {}),
-  ...(isMobile && options.isMobile ? options.isMobile : {}),
-  ...(isIOS && options.isIOS ? options.isIOS : {}),
-  ...(isAndroid && options.isAndroid ? options.isAndroid : {}),
-  ...(isElectron && options.isElectron ? unifyStyles(options.isElectron) : {}),
-})
+type AsStylesCrossPlatform<T> = Readonly<
```
Comment (jakob223) [inline shared/styles/shared.tsx]: What's this type for?

File: shared/common-adapters/text.native.tsx [hunk 6]
```diff
@@ -9,7 +9,7 @@ import {Props, TextType} from './text'
 const modes = ['positive', 'negative']
 
 const styles = Styles.styleSheetCreate(() =>
-  Object.keys(metaData()).reduce<{[key: string]: Styles.StylesCrossPlatform}>(
+  Object.keys(metaData()).reduce<any>(
```
Comment (jakob223) [inline shared/common-adapters/text.native.tsx]: intentionally left as any?

File: shared/desktop/remote/component-loader.desktop.tsx [hunk 7]
```diff
@@ -56,7 +56,7 @@ class RemoteComponentLoader extends React.Component<Props> {
 
   render() {
     return (
-      <div id="RemoteComponentRoot" style={this.props.style || styles.container}>
+      <div id="RemoteComponentRoot" style={this.props.style || (styles.container as any)}>
```
Comment (jakob223) [inline shared/desktop/remote/component-loader.desktop.tsx]: intentionally left as any?

File: shared/common-adapters/icon.native.tsx [hunk 8]
```diff
@@ -19,12 +19,12 @@ type TextProps = {
...
   type: IconType
 }
-
+type Writeable<T> = {-readonly [P in keyof T]: T[P]}
 const Text = React.forwardRef<NativeText, TextProps>((p, ref) => {
-  const style: Styles.StylesCrossPlatform = {}
+  const style: Writeable<Styles.StylesCrossPlatform> = {}
 
   // we really should disallow reaching into style like this but this is what the old code does.
   // TODO change this
```
[the remark this answers is not in the document]
Comment (chrisnojima) [reply shared/common-adapters/icon.native.tsx]: its not safe as the comment on line 29 alludes. there's no ticket as we won't change this in the short term

File: shared/common-adapters/markdown/react.tsx [hunk 9]
```diff
@@ -378,7 +371,7 @@ const serviceOnlyOutput = SimpleMarkdown.reactFor(
             key={state.key}
             allowFontScaling={state.allowFontScaling}
             styleOverride={state.styleOverride}
-            styles={markdownStyles}
+            styles={markdownStyles as any}
```
[the remark this answers is not in the document]
Comment (chrisnojima) [reply shared/common-adapters/markdown/react.tsx]: theres some error i didn't want to stop everything to figure out

File: shared/common-adapters/radio-button.desktop.tsx [hunk 10]
```diff
@@ -36,10 +35,7 @@ const RadioButton = ({disabled, label, onSelect, selected, style}: Props) => (
     onClick={disabled ? undefined : () => onSelect(!selected)}
   >
     <StyledRadio disabled={disabled} selected={selected}>
-      <div
-        // @ts-ignore clash between StylesCrossPlatform and React.CSSProperties
-        style={styles.radio}
-      />
+      <div style={styles.radio as any} />
```
[the remark this answers is not in the document]
Comment (chrisnojima) [reply shared/common-adapters/radio-button.desktop.tsx]: this applies everywhere so its not actually that useful. later on we can remove StylesCrossPlatform but this step was required

File: shared/common-adapters/radio-button.desktop.tsx [hunk 11]
```diff
@@ -21,6 +19,7 @@ const StyledRadio = Styles.styled.div(
     position: 'relative',
     width: RADIOBUTTON_SIZE,
   }),
+  // @ts-ignore
```
[the remark this answers is not in the document]
Comment (chrisnojima) [reply shared/common-adapters/radio-button.desktop.tsx]: basically keeping what we had before where this is untyped. didn't want to fix it now

File: shared/common-adapters/text.native.tsx (hunk 6 above)
Comment (chrisnojima) [reply shared/common-adapters/text.native.tsx]: yeah, didn't want to fix now

File: shared/desktop/remote/component-loader.desktop.tsx (hunk 7 above)
Comment (chrisnojima) [reply shared/desktop/remote/component-loader.desktop.tsx]: yup

File: shared/styles/shared.tsx (hunk 5 above)
Comment (chrisnojima) [reply shared/styles/shared.tsx]: this is what platformStyles returns below. It casts the keys as the values of cross platform so things that consume stylescrossplatform can accept the styles. else if you use this helper you can't use the styles anymore. later we can ditch styles cross platform and have the actual styles

File: shared/styles/css.d.ts [hunk 12]
```diff
@@ -18,18 +122,21 @@ type _StylesCrossPlatformOverride = {
...
-    : never
-}
+  }
+>
 
 export type StylesCrossPlatform = StyleProp<_StylesCrossPlatform>
 
-export type StylesCrossPlatformWithSomeDisallowed<D> = StyleProp<_StylesCrossPlatform & D>
+export type _CustomStyles<K extends string, C> = Omit<_StylesCrossPlatform, K> & Readonly<C>
+export type CustomStyles<K extends string, C> = StyleProp<_CustomStyles<K, C>>
```
[the remark this answers is not in the document]
Comment (chrisnojima) [reply shared/styles/css.d.ts]: please suggest a name

[the remark this answers is not in the document]
Comment (chrisnojima) [reply shared/styles/css.d.ts]: we can rename it after this is merged

[the remark this answers is not in the document]
Comment (jakob223) [reply shared/styles/css.d.ts]: Maybe have two types, `StylesWithout` and `StylesWithReplaced` where the first one is a special case of the second one?

**Pull request merged**